examples/c/builtin_echo.c

```diff
@@ -0,0 +1,199 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+
+#include "teaclave_client_sdk.h"
+#include <stdio.h>
+#include <string.h>
+
+#define BUFFER_SIZE 4086
+
+const char *authentication_service_address = "localhost:7776";
+const char *frontend_service_address = "localhost:7777";
+const char *enclave_info_path = "../../release/services/enclave_info.toml";
+#ifdef DCAP
+const char *as_root_ca_cert_path = "../../keys/dcap_root_ca_cert.pem";
+#else
+const char *as_root_ca_cert_path = "../../keys/ias_root_ca_cert.pem";
+#endif
+const char *user_id = "test_id";
+const char *user_password = "test_password";
+
+const char *register_function_request_serialized =
+    "{"
+    "    \"request\": \"register_function\","
+    "    \"name\": \"builtin-echo\","
+    "    \"description\": \"Native Echo Function\","
+    "    \"executor_type\": \"builtin\","
+    "    \"public\": true,"
+    "    \"payload\": [],"
+    "    \"arguments\": ["
+    "        \"message\""
+    "    ],"
+    "    \"inputs\": [],"
+    "    \"outputs\": []"
+    "}";
+
+const char *create_task_request_serialized =
+    "{"
+    "    \"request\": \"create_task\","
+    "    \"function_id\": \"%s\","
+    "    \"function_arguments\": \"{\\\"message\\\": \\\"Hello, "
+    "Teaclave!\\\"}\","
+    "    \"executor\": \"builtin\","
+    "    \"inputs_ownership\": [],"
+    "    \"outputs_ownership\": []"
+    "}";
+
+int login(char *token, size_t *token_len) {
+    int ret = 0;
+
+    AuthenticationClient *authentication_client =
+        teaclave_connect_authentication_service(authentication_service_address,
+                                                enclave_info_path,
+                                                as_root_ca_cert_path);
+    if (authentication_client == NULL) {
+        fprintf(stderr,
+                "[-] Failed to connect to the authentication service.\n");
+        ret = 1;
+        goto bail;
+    }
+
+    ret = teaclave_user_register(authentication_client, user_id, user_password);
+    if (ret != 0) {
+        fprintf(stderr, "[-] Failed to register user.\n");
+        fprintf(stderr, "[-] Maybe `%s' already exists. Continue. \n", user_id);
+    }
+
+    ret = teaclave_user_login(authentication_client, user_id, user_password,
+                              token, token_len);
+    if (ret != 0) {
+        fprintf(stderr, "[-] Failed to login.\n");
+        goto bail;
+    }
+    printf("[+] token: %s\n", token);
+
+bail:
+    if (authentication_client) {
+        ret = teaclave_close_authentication_service(authentication_client);
+        if (ret != 0) {
+            fprintf(stderr,
+                    "[-] Failed to close the authentication service client.\n");
+        }
+    }
+
+    return ret;
+}
+
+int main() {
+    int ret = 0;
+    char *token = (char *)calloc(BUFFER_SIZE, sizeof(char));
```

Comment:
Wouldn't it be nice to encourage stack allocation instead? Especially since you're using a fixed size buffer anyhow.

Reply:
Yes, I have replaced them as stack allocated buffers.